README.md

```diff
@@ -1,15 +1,151 @@
+
+The smart contracts can easily be run by deploying them to testnet. Choose a contract and follow these instructions. 
+
+#### Build
+First the smart contract must be built with this command from the `increment` contract’s root folder:
+
+```
+cd increment
+stellar contract build
+```
+
+A `.wasm` file will be outputted in the target directory, at `target/wasm32-unknown-unknown/release/soroban_increment_contract.wasm`. The `.wasm` file is the built contract.
+
+#### Deploy
+The WASM file can now be deployed to the testnet by running this command:
+
+```
+stellar contract deploy \
+  --wasm target/wasm32-unknown-unknown/release/soroban_increment_contract.wasm \
+  --source alice \
+  --network testnet \
+  --alias increment_contract
+```
+
+When the smart contract has been successfully deployed, the command will return the contract’s ID (e.g. CACDYF3CYMJEJTIVFESQYZTN67GO2R5D5IUABTCUG3HXQSRXCSOROBAN). This ID can be used to invoke the contract, but since an alias is added, the alias can be used for invoking the contract as well.
+
+#### Invoke
+Now the contract is on testnet, it can be invoked. For the increment contract there’s only one function to invoke, and that’s the increment() function. Look at the code for the other contracts to see which function to invoke as every example contract is different.
+
+Run this command to invoke the increment contract (the added alias is used as the contract ID):
+
+```
+stellar contract invoke \
+  --id increment_contract \
+  --source alice \
+  --network testnet \
+  -- \
+  increment 
+```
+
+The contract will return 1 the first time it’s run, run it again and see the returned value is being incremented.
+
+## Testing
+Each of the example smart contracts also has a test file that has test cases for each of the features of the smart contracts. The test will just return a pass/fail result, but it’s a convenient way to check if the code works, without deploying and invoking the contract manually. 
+
+From the root of the contract (e.g. `increment`) run this command:
+
+```
+cargo test
+```
+
+Some examples may contain multiple contracts and require contracts to be built before the test can be run. See the individual example contracts for details.
+
+## Licence
+The example smart contracts are licensed under the Apache 2.0 license. See the LICENSE file for details.
+
+## Contributions
+Contributions are welcome, please create a pull request with the following information: 
+
+- Explain the changes/additions you made
+- Why are these changes/additions needed or relevant?
+- How did you solve the problem, or created the suggested feature?
+- Have your changes/additions been thoroughly tested?
+
+## Disclaimer
+These implementations have not been tested or audited. They are likely to have significant errors and security vulnerabilities. They should not be relied on for any purpose. Please refer to the license for more information.
+
```

Comment:
I think this should move to the top, and use a warning box. I suggested an edit above.
```suggestion
```